src/main/java/org/openrewrite/staticanalysis/TernaryOperatorsShouldNotBeNested.java

```diff
@@ -0,0 +1,92 @@
+package org.openrewrite.staticanalysis;
+
+import java.time.Duration;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Set;
+
+import org.openrewrite.ExecutionContext;
+import org.openrewrite.Recipe;
+import org.openrewrite.Tree;
+import org.openrewrite.TreeVisitor;
+import org.openrewrite.internal.lang.Nullable;
+import org.openrewrite.java.JavaTemplate;
+import org.openrewrite.java.JavaVisitor;
+import org.openrewrite.java.tree.Expression;
+import org.openrewrite.java.tree.J;
+import org.openrewrite.java.tree.JavaSourceFile;
+import org.openrewrite.java.tree.Space;
+import org.openrewrite.java.tree.Statement;
+import org.openrewrite.marker.Markers;
+
+
+public class TernaryOperatorsShouldNotBeNested extends Recipe {
+    @Override
+    public String getDisplayName() {
+        return "Ternary operators should not be nested";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Just because you can do something, doesn’t mean you should, and that’s the case with nested ternary operations. " +
+                "Nesting ternary operators results in the kind of code that may seem clear as day when you write it, " +
+                "but six months later will leave maintainers (or worse - future you) scratching their heads and cursing.\n\n" +
+                "Instead, err on the side of clarity, and use another line to express the nested operation as a separate statement.";
```

Comment:
This gave me a good chuckle. At first I thought it's not as inline with our other recipe descriptions, but it turns out it's a direct copy of the description over on: https://sonarsource.atlassian.net/browse/RSPEC-3358

Now I'm wondering if we should rewrite this to be a bit shorter and more in line, and potentially not use the description one-to-one as seen on an external source. Something like:
```suggestion
        return "Nested ternary operators can be hard to read quickly. Prefer simpler constructs for improved readability.";
```